exonum_client/protoc.py

```diff
@@ -42,6 +46,26 @@ def _modify_file(path: str, modules: List[str]) -> None:
                     line = line.replace("import {}_pb2 ".format(module), "from . import {}_pb2 ".format(module))
                 file_out.write(line)
 
+    def _ensure_protoc_version(self) -> None:
+        """Checks that installed protoc has sufficient version."""
+        protoc_args = [self._protoc_path, "--version"]
+        version_stdout = subprocess.run(protoc_args, stdout=subprocess.PIPE, check=True).stdout
+
+        # Convert bytestring to string and split into words.
+        # b'libprotoc 3.7.1\n' => ['libprotoc', '3.7.1']
+        version = version_stdout.decode("utf-8").strip().split()
+
+        # Expected output is like 'libprotoc 3.7.1'
+        if len(version) != 2 or not re.match(r"\d+.\d+.\d+", version[1]):
+            raise RuntimeError(f"Unexpected output on resolving protoc version: {version}")
```

Comment:
Maybe `version[1]`?

Reply:
Nope. This error will be raised if `len(version) != 2`, e.g. if `protoc` provided output like "version1.2.3" for some reason, and thus we want to show the whole output.

Reply:
Ok, I see. Maybe it is better to print the original output then, instead of the list generated by split? It's not a big deal but just to be consistent )

Reply:
It will then require to either print a byte string (and pylint doesn't like that idea), or introduce a new variable with decoded to utf-8, but not yet splitted string. Since this exception is unlikely to ever happen (since it requires protoc to be installed, but output something weird at the same time) and inserted "just in case", I think  the current implementation is OK.